ProcessLib/ThermoMechanics/ThermoMechanicsFEM.h

```diff
@@ -0,0 +1,550 @@
+
+    virtual std::vector<double> const& getIntPtEpsilonXX(
+        std::vector<double>& cache) const = 0;
+
+    virtual std::vector<double> const& getIntPtEpsilonYY(
+        std::vector<double>& cache) const = 0;
+
+    virtual std::vector<double> const& getIntPtEpsilonZZ(
+        std::vector<double>& cache) const = 0;
+
+    virtual std::vector<double> const& getIntPtEpsilonXY(
+        std::vector<double>& cache) const = 0;
+
+    virtual std::vector<double> const& getIntPtEpsilonXZ(
+        std::vector<double>& cache) const = 0;
+
+    virtual std::vector<double> const& getIntPtEpsilonYZ(
+        std::vector<double>& cache) const = 0;
+};
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          int DisplacementDim>
+class ThermoMechanicsLocalAssembler
+    : public ThermoMechanicsLocalAssemblerInterface
+{
+public:
+    using ShapeMatricesType =
+        ShapeMatrixPolicyType<ShapeFunction, DisplacementDim>;
+
+    // Types for displacement.
+    // (Higher order elements = ShapeFunction).
+    using ShapeMatrices = typename ShapeMatricesType::ShapeMatrices;
+    using BMatricesType = BMatrixPolicyType<ShapeFunction, DisplacementDim>;
+
+    using NodalForceVectorType = typename BMatricesType::NodalForceVectorType;
+    using RhsVector = typename ShapeMatricesType::template VectorType<
+        ShapeFunction::NPOINTS + ShapeFunction::NPOINTS * DisplacementDim>;
+    using JacobianMatrix = typename ShapeMatricesType::template MatrixType<
+        ShapeFunction::NPOINTS + ShapeFunction::NPOINTS * DisplacementDim,
+        ShapeFunction::NPOINTS + ShapeFunction::NPOINTS * DisplacementDim>;
+
+    ThermoMechanicsLocalAssembler(ThermoMechanicsLocalAssembler const&) =
+        delete;
+    ThermoMechanicsLocalAssembler(ThermoMechanicsLocalAssembler&&) = delete;
+
+    ThermoMechanicsLocalAssembler(
+        MeshLib::Element const& e,
+        std::size_t const /*local_matrix_size*/,
+        bool const is_axially_symmetric,
+        unsigned const integration_order,
+        ThermoMechanicsProcessData<DisplacementDim>& process_data)
+        : _process_data(process_data),
+          _integration_method(integration_order),
+          _element(e)
+    {
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        _ip_data.reserve(n_integration_points);
+        _secondary_data.N.resize(n_integration_points);
+
+        auto const shape_matrices =
+            initShapeMatrices<ShapeFunction, ShapeMatricesType,
+                              IntegrationMethod, DisplacementDim>(
+                e, is_axially_symmetric, _integration_method);
+
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            // displacement (subscript u)
+            _ip_data.emplace_back(*_process_data.material);
+            auto& ip_data = _ip_data[ip];
+            ip_data.integration_weight =
+                _integration_method.getWeightedPoint(ip).getWeight() *
```

Comment:
Missing `_integration_method.integralMeasure`? and change B matrix for axisymmetric problem.